speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/FileDataStorageTests.cpp

```diff
@@ -35,31 +81,327 @@ inline std::string GetTimeAsString(time_t p_date)
     return buffer;
 }
 
+/// @brief Generate a random DecisionTuple containing the decisions given
+/// @param p_random Random generator to use for determining values
+/// @param p_doSteer Whether the DecisionTuple should include a Steer Decision
+/// @param p_doBrake Whether the DecisionTuple should include a Brake Decision
+/// @param p_doAccel Whether the DecisionTuple should include a Accel Decision
+/// @param p_doGear Whether the DecisionTuple should include a Gear Decision
+/// @param p_doLights Whether the DecisionTuple should include a Lights Decision
+/// @return A random DecisionTuple containing the decisions as determined by the parameters
+DecisionTuple GenerateDecisions(Random& p_random, bool p_doSteer, bool p_doBrake, bool p_doAccel, bool p_doGear, bool p_doLights)
+{
+    // Generate a random decision based on parameters
+    DecisionTuple decisions;
+    if (p_doSteer) decisions.SetSteer(p_random.NextFloat());
+    if (p_doBrake) decisions.SetBrake(p_random.NextFloat());
+    if (p_doAccel) decisions.SetAccel(p_random.NextFloat());
+    if (p_doGear) decisions.SetGear(p_random.NextInt());
+    if (p_doLights) decisions.SetLights(p_random.NextBool());
+    return decisions;
+}
+
+/// @brief Write to a stream the data expected to be seen in a file when decisions are stored
+/// @param decisions The tuple containing the decision data
+/// @param expected The stream containing the expected values of the file
+/// @param p_doSteer Whether a Steer decision  should be written
+/// @param p_doBrake Whether a Brake decision  should be written
+/// @param p_doAccel Whether a Accel decision  should be written
+/// @param p_doGear Whether a Gear decision  should be written
+/// @param p_doLights Whether a Lights decision  should be written
+void WriteExpectedDecisions(DecisionTuple& p_decisions, std::ostream& p_expected, bool p_doSteer, bool p_doBrake, bool p_doAccel, bool p_doGear, bool p_doLights)
+{
+    p_expected << "Decisions\n";
+    if (p_doSteer)
+    {
+        p_expected << "SteerDecision\n"
+                   << std::to_string(p_decisions.GetSteer()) << "\n";
+    }
+    if (p_doBrake)
+    {
+        p_expected << "BrakeDecision\n"
+                   << std::to_string(p_decisions.GetBrake()) << "\n";
+    }
+    if (p_doAccel)
+    {
+        p_expected << "AccelDecision\n"
+                   << std::to_string(p_decisions.GetAccel()) << "\n";
+    }
+    if (p_doGear)
+    {
+        p_expected << "GearDecision\n"
+                   << std::to_string(p_decisions.GetGear()) << "\n";
+    }
+    if (p_doLights)
+    {
+        p_expected << "LightsDecision\n"
+                   << std::to_string(p_decisions.GetLights()) << "\n";
+    }
+    p_expected << "NONE\n";
```

Comment:
Change all '\n' characters with std::endl as enters are different on windows and linux